src/dune_pkg/opam_solver.ml

```diff
@@ -810,7 +810,13 @@ let reject_unreachable_packages =
             [ "name", Package_name.to_dyn name ]
         | Some (pkg : Lock_dir.Pkg.t), None -> Some (List.map pkg.depends ~f:snd)
         | None, Some (pkg : Local_package.For_solver.t) ->
-          let formula = pkg.dependencies |> Dependency_formula.to_filtered_formula in
+          (* remove Dune from the formula as we remove it from solutions *)
+          let formula =
```

Comment:
This is mostly a question to understand the behavior. Will it keep the constraint on Dune somewhere, even if we remove it from the formula? This is just to understand the way it is computed underneath.

Reply:
This is just to check which packages are reachable, it is not what is used to compute the solution. As you can see in CI, the `implicit-dune-deps.t` test that checks the dune version against a constraint is passing.